modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PreferenceProperties.java

```diff
@@ -102,14 +139,23 @@ public void setAccentColor(Color color) {
         accentColor.setValueOverride(color);
     }
 
-    public void update(Map<String, ChangedValue> changedPreferences, Map<String, String> platformKeyMappings) {
+    public void update(Map<String, ChangedValue> changedPreferences,
+                       Map<String, PreferenceMapping<?>> platformKeyMappings) {
+        outerLoop:
         for (Map.Entry<String, ChangedValue> entry : changedPreferences.entrySet()) {
-            String key = platformKeyMappings.get(entry.getKey());
-            if (key != null) {
-                for (ColorProperty colorProperty : allColors) {
-                    if (colorProperty.getName().equals(key)) {
-                        updateColorProperty(colorProperty, entry.getValue().newValue());
-                        break;
+            PreferenceMapping<?> mapping = platformKeyMappings.get(entry.getKey());
+            if (mapping != null) {
```

Comment:
this code seems rather inefficient:

why are we doing linear search?
also why are we searching through colors when, for example, the changed property is a boolean?

is it possible to do a simple hash map lookup instead?

Reply:
Good idea, this has also removed quite a bit of code.